src/Voron/Impl/LowLevelTransaction.cs

```diff
@@ -96,6 +96,8 @@ public void Reset()
 
         public event Action<LowLevelTransaction> LastChanceToReadFromWriteTransactionBeforeCommit;
 
+        public long TransactionSizeInPages => NumberOfModifiedPages + AdditionalMemoryUsageSize.GetValue(SizeUnit.Bytes) / Constants.Storage.PageSize;
```

Comment:
This is weird, here you could easily calculate that value 

```
public Size TransactionSize => new Size(NumberOfModifiedPages * Constants.Storage.PageSize, Bytes) + AdditionalMemoryUsageSize;
```

then in SHardedDcoumentsStorage you would have value in 'Size', which is more natural when processing thresholds than doing calculations in 'Pages'

Reply:
done